### PR TITLE
Support PHP 8.1

### PR DESCRIPTION
**Pull request by paul-m**

### CODE REVIEW

**Comment by paul-m** [comment]
There was a single test and it was failing, because it was a bad test.

I expanded the tests so they give us 100% coverage, and some generic easily-testable cases as well.

**Review by jastraat — COMMENTED** [review]
Do we want Codeclimate on this?

File: phpcs.xml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+    <file>src</file>
+    <file>tests</file>
+    <file>rector.php</file>
+
+    <rule ref="Drupal">
+        <exclude name="Drupal.Commenting.FunctionComment.Missing"/>
+        <exclude name="Drupal.Commenting.DocComment.MissingShort"/>
+    </rule>
+    <rule ref="DrupalPractice"/>
```
Comment (jastraat) [inline phpcs.xml:15]: @paul-m The other DKAN libraries were using PSR1 and PSR2. Just curious as to why this one is using the DrupalPractice instead?

Comment (jastraat) [reply phpcs.xml:15]: Per Paul, "pdlt looks like Drupal rather than PSR1/2, so I enforced that. " So we're good :)

**Review by jastraat — APPROVED** [review]
All matrix automated tests for PHP 7.4, 8.0, and 8.1 passing (as well as with prefer lowest setting). Codesniffer with DrupalPractice rules. And Codeclimate. Looks good!

**Pull request merged**